[docs] Update how-to-build-compiler (#15500)
Samsung/ONE: docs/howto/how-to-build-compiler.md

```
will build release version in `build/release` folder.

> ## Local install and command line tests

Reviewer: ```suggestion
## Local install and command-line tests
```

> Local install test is to make an local installation in `build/install` path,

Reviewer: ```suggestion
Local install test is to make a local installation in `build/install` path,
```

> check the build artifacts and run compiler command line tests of `onecc`

Reviewer: ```suggestion
check the build artifacts and run compiler command-line tests of `onecc`
```

> Local install test is to make an local installation in `build/install` path,
> check the build artifacts and run compiler command line tests of `onecc`
> and others.

Reviewer: ```suggestion
This guide explains how to perform a local installation to the `build/install` path,  
verify the build artifacts, and run command-line tests for tools such as `onecc`.
```

### Prepare onnx2circle

`one-import-onnx` tool now uses `onnx2circle` tool from `circle-mlir`.

> `onnx2cicle` tool can be prepared with (1) build from source or (2) install

Reviewer: ```suggestion
`onnx2circle` tool can be prepared with (1) build from source or (2) install
```

from [launchpad.net](https://launchpad.net/~circletools).

Please read [README.md](../circle-mlir/README.md) for details how to build.

As `circle-mlir/externals` uses lots of storage(about 150~200GB) and time,

> we recommand to use build with Docker image as described in above README file.

Reviewer 2: recommand -> recommend

> As `circle-mlir/externals` uses lots of storage(about 150~200GB) and time,
> we recommand to use build with Docker image as described in above README file.

Reviewer: ```suggestion
> ⚠️ Note: The `circle-mlir/externals` directory consumes a large amount of disk space (150~200 GB) and takes significant time to build.  
> It is recommended to build using the Docker image, as described in the README.
```

To use Debian package from [launchpad.net](https://launchpad.net/~circletools),
```
sudo apt-get update
sudo apt-get install software-properties-common
sudo -E add-apt-repository ppa:circletools/nightly
sudo apt-get install onnx2circle
```
- this will install `onnx2circle` tool to `/usr/share/circletools` folder

`one-cmds/CMakeLists.txt` will install `onnx2circle` from (1) or (2) to

> `buid/install` as conversion tool of `one-import-onnx` command.

Reviewer: ```suggestion
`build/install` as conversion tool of `one-import-onnx` command.
```

### How to make local installation
```
NNAS_BUILD_PREFIX=build/setup \
BUILD_TYPE=Release \
./nnas create-package --preset 20230907 --prefix build/install
```
This will configure and build necessary modules and install files in
`build/install` path.

> ### Local command line tests

Reviewer: ```suggestion
### Local command-line tests
```

> To run local command line tests, python virtual envirnment is necessary with

Reviewer: ```suggestion
To run local-command line tests, python virtual environment is necessary with
```

test models.

Prepare python virtual environment:
```
pushd build/install/bin
bash ./one-prepare-venv
popd
```

Download test models:
```
pushd build/install/test
bash ./prepare_test_materials.sh
popd
```

> Run command line tests:

Reviewer: ```suggestion
Run command-line tests:
```
